File: DotNetProducer/SendReservation/Models/Reservation.cs
namespace SendReservation.Models
{
    public class Reservation
    {
        public int hotelId { get; set; }
#pragma warning disable CS8618 // Non-nullable field must contain a non-null value when exiting constructor. Consider declaring as nullable.
        public string checkIn { get; set; } // ISO 8601 date
        public string checkOut { get; set; } // ISO 8601 date 
        public int roomNo { get; set; }
        public string customerName { get; set; }
        public string customerEmail { get; set; }
        public string customerAddress { get; set;}
    }
}

File: HandleReservation/src/reservation.ts
import { Schema } from 'mongoose'

export interface Reservation {
    hotelId: string
    checkIn: string
    checkOut: string
    roomNo: number
    customerName: string
    customerEmail:string
    customerAddress:string
}

export const schema = new Schema<Reservation>({
    hotelId: { type: String, required: true },
    checkIn: { type: String, required: true },
    checkOut: { type: String, required: true },
    roomNo: { type: Number, required: true },
    customerName: { type: String, required: true },
    customerEmail:{ type: String, required: true },
    customerAddress:{ type: String, required: true },
})